starter-code/scripts/models/article.js

```diff
@@ -45,21 +45,28 @@ Article.loadAll = function(inputData) {
 /* This function below will retrieve the data from either a local or remote
  source, process it, then hand off control to the View: */
 Article.fetchAll = function() {
-  if (localStorage.hackerIpsum) {
+  if (localStorage.hackerIpsumObjects) {
     /* When our data is already in localStorage:
     1. We can process and load it,
     2. Then we can render the index page.  */
+    var localStorageData = JSON.parse(localStorage.getItem('hackerIpsumObjects'));
+    Article.loadAll(localStorageData);
+    articleView.renderIndexPage();
   } else {
     /* Without our localStorage in memory, we need to:
     1. Retrieve our JSON file with $.getJSON
       1.a Load our json data
       1.b Store that data in localStorage so that we can skip the server call next time,
       1.c And then render the index page.*/
+    $.getJSON('data/hackerIpsum.json')
+    .done(function(data){
+      localStorage.setItem('hackerIpsumObjects', JSON.stringify(data));
+      Article.loadAll(data);
+      articleView.renderIndexPage();
```

Comment:
Awesome job ensuring that the asynchronous AJAX call has completed successfully before processing the data (setting it to local storage, instantiating our objects, creating our array, and rendering the page).